lib/kmts.rb

```diff
@@ -191,13 +191,23 @@ def generate_query(type, data, id = nil)
       data.update('_k' => @key)
       data.update '_d' => 1 if data['_t']
       data['_t'] ||= Time.now.to_i
-      
+
       unsafe = Regexp.new("[^#{URI::REGEXP::PATTERN::UNRESERVED}]", false, 'N')
       
       data.inject(query) do |query,key_val|
         query_arr <<  key_val.collect { |i| URI.escape(i.to_s, unsafe) }.join('=')
       end
       query = '/' + type + '?' + query_arr.join('&')
+
+      begin
+        unless data['_t'].to_s =~ /\d{10}/ and data['_t'].to_i >= 0
```

Comment:
I could! I thought that since its only going to be used right here, bringing it out would be overkill.

Totally agree about the `raise` here, I am not happy about it but I am working with the patterns in here.

Good calls on the rest, thanks for spotting it!